lib/inspec/runner.rb

```diff
@@ -57,6 +57,12 @@ def initialize(conf = {})
         RunnerRspec.new(@conf)
       end
 
+      if @conf[:waiver_file]
+        waivers = cleanse_waivers @conf.delete(:waiver_file)
+        @conf[:input_file] ||= []
+        @conf[:input_file].concat waivers
```

Comment:
So, we have a choice here - if waivers are going to be YAML files, very similar to input files,  this will work fine, and we need not pre-process them at all - `cleanse_waivers` won't be needed, AFAIK. The InputRegistry expects `:input_file` to be an array of YAML pathnames, and it will try to read them as such.

So if we choose any other format, we need to either read it ourselves in the InputRegistry, or read it here (which would be surprising, I think).